bootstrap/controller/commonapi.go

```diff
@@ -0,0 +1,223 @@
+// It returns a response as specified by the API swagger in the openapi directory
+func (c *CommonController) Ping(writer http.ResponseWriter, request *http.Request) {
+	response := commonDTO.NewPingResponse(c.serviceName)
+	c.sendResponse(writer, request, common.ApiPingRoute, response, http.StatusOK)
+}
+
+// Version handles the request to /version endpoint. Is used to request the service's versions
+// It returns a response as specified by the API swagger in the openapi directory
+func (c *CommonController) Version(writer http.ResponseWriter, request *http.Request) {
+	var response interface{}
+	if c.version.sdkVersion != "" {
+		response = commonDTO.NewVersionSdkResponse(c.version.serviceVersion, c.version.sdkVersion, c.serviceName)
+	} else {
+		response = commonDTO.NewVersionResponse(c.version.serviceVersion, c.serviceName)
+	}
+	c.sendResponse(writer, request, common.ApiVersionRoute, response, http.StatusOK)
+}
+
+// Config handles the request to /config endpoint. Is used to request the service's configuration
+// It returns a response as specified by the V2 API swagger in openapi/common
+func (c *CommonController) Config(writer http.ResponseWriter, request *http.Request) {
+	var fullConfig interface{}
+
+	if c.config.customConfig == nil {
+		// case of no custom configs
+		fullConfig = c.config.configuration
+	} else {
+		// create a struct combining the common configuration and custom configuration sections
+		fullConfig = struct {
+			interfaces.Configuration
+			CustomConfiguration interfaces.UpdatableConfig
+		}{
+			c.config.configuration,
+			c.config.customConfig,
+		}
+	}
+
+	response := commonDTO.NewConfigResponse(fullConfig, c.serviceName)
+	c.sendResponse(writer, request, common.ApiVersionRoute, response, http.StatusOK)
+}
+
+// AddSecret handles the request to the /secret endpoint. Is used to add EdgeX Service exclusive secret to the Secret Store
+// It returns a response as specified by the API swagger in the openapi directory
+func (c *CommonController) AddSecret(writer http.ResponseWriter, request *http.Request) {
+	defer func() {
+		_ = request.Body.Close()
+	}()
+
+	secretRequest := commonDTO.SecretRequest{}
+	err := json.NewDecoder(request.Body).Decode(&secretRequest)
+	if err != nil {
+		c.lc.Infof("%v", err.Error())
+		c.sendError(writer, request, errors.KindContractInvalid, "JSON decode failed", err, common.ApiSecretRoute, "")
+		return
+	}
+
+	err = addSecret(c.dic, secretRequest)
+	if err != nil {
+		c.sendError(writer, request, errors.Kind(err), err.Error(), err, common.ApiSecretRoute, secretRequest.RequestId)
+		return
+	}
+
+	response := commonDTO.NewBaseResponse(secretRequest.RequestId, "", http.StatusCreated)
+	c.sendResponse(writer, request, common.ApiSecretRoute, response, http.StatusCreated)
+
+}
+
+// sendResponse puts together the response packet for the APIs
+func (c *CommonController) sendResponse(
+	writer http.ResponseWriter,
+	request *http.Request,
+	api string,
```

Comment:
remove `api` in this function because it's for error log only
just log the response type, and it could provide enough indication